src/components/SidePanel.tsx

```diff
@@ -1,16 +1,44 @@
-import * as React from 'react';
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import { Checkbox, Tooltip } from 'antd';
+
+import { HiddenOrHighlightedState, TOOLTIP_COLOR } from '../constants';
+import { VisibilityContext } from '../AgentVisibilityContext';
 
 import '../../css/side_panel.css';
 
-// TODO placeholder, side panel will receive many props when rendering checkboxes
-interface SidePanelProps {}
+const SidePanel: React.FunctionComponent = (): JSX.Element => {
+  const { hideOrRevealAllAgents, allAgentsHiddenState } =
+    useContext(VisibilityContext);
+
+  const { Inactive, Active, Indeterminate } = HiddenOrHighlightedState;
+
+  const tooltipText = {
+    [Active]: 'Show',
+    [Inactive]: 'Hide',
+    [Indeterminate]: 'Show',
+  };
 
-const SidePanel: React.FunctionComponent<SidePanelProps> = (
-  props: SidePanelProps
-): JSX.Element => {
   return (
     <div className="sp-container">
       <div className="agent-title">Agents</div>
+      <div className="checkboxtree">
+        <div className="item-row">
+          <Tooltip
+            placement="right"
+            title={tooltipText[allAgentsHiddenState]}
+            color={TOOLTIP_COLOR}
+          >
+            <Checkbox
+              className={classNames('checkbox', 'check-all')}
+              indeterminate={allAgentsHiddenState === Indeterminate}
+              checked={allAgentsHiddenState === Inactive}
```

Comment:
Is this inverted? When `allAgentsHiddenState` is set to `Active`, does that mean that all agents are hidden or visible?

Reply:
I think this points to a greater ambiguity between `hiddenState` and `visibility`. I would expect `visibility=Active` to mean that objects are visible, while `hiddenState=Active` is more ambiguous and could differ in interpretation by developer.

Reply:
I'm making the changes you've suggested around changing `ActiveState` and `setAllAgentVisibility` but it doesn't resolve this issue...
What do you think of this:
```
const visible = allAgentsHiddenState === Inactive; // agents are visible when hiding is inactive
```
and then
```
checked={visible}

Reply:
Yeah, I think this would definitely make it more readable. Thank you for thinking about this!

Reply:
Done. :)

Reply:
Just in response to your comment @interim17 about the selection interface: The way to think about it is it reflects actions by the user. By default, all agents are visible and unhighlighted. So the deviation is to be either hidden or highlighted. Just because one is making something "more visible" and one is making it invisible, I don't think you need to think of those as inversions. They are both just changes from the default.